modules/gog.py

```diff
@@ -0,0 +1,202 @@
+import copy
+import logging
+from datetime import datetime
+import time
+
+from gevent import monkey
+
+monkey.patch_all()
+
+import requests
+
+from modules import utils  # noqa: E402
+from modules.models import App, Cache, Result  # noqa: E402
+
+import json
+
+class GOGAppFilter:
+    # None = null (default release branch)
+    def __init__(self, id, filter=None):
+        self.id = id
+        self.filter = filter
+
+    def __str__(self):
+        # GOG uses null for main branch
+        # null != "null" so we add quotes around strings to avoid confusion
+        return self.id + ":" + ("null" if  self.filter is None else "\"" + self.filter + "\"") 
+
+
+class GOG:
+    def __init__(self, app_ids, notifier, ignore_first):
+        self.logger = logging.getLogger(__name__)
+        self.timestamp = None
+
+        utils.create_directory("./cache/gog")
+        self.cache = Cache(
+            "./cache/gog/latest_data.json",
+            "./cache/gog/old_data.json",
+            "./cache/gog/tmp_data.json",
+            "./cache/gog/latest_result.json",
+        )
+
+        self.old_result = {}
+        self.new_result = utils.load_json_as_dict(self.cache.result)
+
+        self.ignore_first = ignore_first
+
+        self.apps = []
+
+        for _app_id in app_ids:
+            _app = _app_id.split(":")
+            new_filter = GOGAppFilter(id=_app[0])
+            if len(_app) != 1:
+                new_filter = GOGAppFilter(id=_app[0], filter=_app[1])
+            self.apps.append(new_filter)
+
+            # null != "null"
+            _app_id = str(new_filter)
+            if _app_id in self.new_result:
+                # de-JSON
+                self.new_result[_app_id] = Result(
```

Comment:
I think the bug for multible branch traking that you found exists for GOG too.
It's okay to leave it as is, but this is a comment just in case.

Reply:
I double-checked this and it doesn't seem to have the bug 👍